src/splinterdb.c

```diff
@@ -137,8 +142,8 @@ splinterdb_validate_app_data_config(const data_config *cfg)
  *-----------------------------------------------------------------------------
  */
 static platform_status
-splinterdb_init_config(const splinterdb_config *kvs_cfg, // IN
-                       splinterdb              *kvs      // OUT
+splinterdb_init_config(splinterdb_config *kvs_cfg, // IN
```

Comment:
This change of taking out `const` qualifier can come out. I had it like-this in a prior dev version, but now I see that in this final version, it's only read-only. Will change this back.

Reply:
Revisit & fix this ... this came up on further discussions with Rob.

Reply:
Resolved. Reworked interfaces to re-introduce `const splinterdb_config *kvs_cfg`.

Cleaned up backout code. Added new unit test case.